src/execution_plan/ops/op_unwind.c

```diff
@@ -32,36 +37,88 @@ OpBase *NewUnwindOp(uint record_idx, AR_ExpNode **exps) {
 }
 
 OpResult UnwindInit(OpBase *opBase) {
+	OpUnwind *unwind = (OpUnwind *)opBase;
+
+	// check if there are children - if not, add static record
+	if(!unwind->op.childCount)
+		unwind->currentRecord = Record_New(opBase->record_map->record_len);
+
+	// check for modifiers in the AR_EXP for static or dynamic list
+	rax *modifies = raxNew();
+	AR_EXP_CollectEntityIDs(unwind->expressions, modifies);
+	// if there aren't any modifiers - list is static
+	if(!raxSize(modifies)) {
+		// set the list
+		unwind->list = AR_EXP_Evaluate(unwind->expressions, NULL);
+		assert(unwind->list.type == T_ARRAY);
+		unwind->isStatic = true;
+		unwind->listIdx = 0;
+	}
+
 	return OP_OK;
 }
 
+// try to generate new value to return
+// NULL will be returned if dynamic list is not evaluted (listIdx = UNDEFIND)
+// or in case of the current list fully returned its memebers
+Record _handoff(OpUnwind *op) {
+	// if there is a new value ready, return it
+	if(op->listIdx < array_len(op->list.array)) {
```

Comment:
I think we should introduce the bare minimum functionality to work with SI_Array, i hate revealing the under line arr value